src/components/pages/Publish/index.tsx

```diff
@@ -81,16 +103,52 @@ export default function PublishPage({
     }
   }
 
+  async function handleAlgorithmSubmit(
+    values: Partial<AlgorithmPublishForm>,
+    resetForm: () => void
+  ): Promise<void> {
+    const metadata = transformPublishAlgorithmFormToMetadata(values)
+
+    try {
+      Logger.log('Publish Algorithm with ', metadata)
+
+      const ddo = await publish((metadata as unknown) as Metadata, 'access')
```

Comment:
this line is totally the tricky part cause we map something else in the UI to something different in our tech stack. This concerns the `optional checkbox "Keep my algorithm private"` part from #341 which implies:

- user sets it to public → we set service type === 'access'
- user sets it to private → we set service type === 'compute'

Reply:
ok got it, wanted to ask after the planning what 'keep my algorithm private implies', but it all clear now

Reply:
For now we can have a simple condition  `await publish((metadata as unknown) as Metadata, algoPrivacy ==='public'? 'access': 'compute')`